src/routes/write-exercise/WriteExerciseScreen.tsx

```diff
@@ -101,6 +105,22 @@ const WriteExerciseScreen = ({ route, navigation }: WriteExerciseScreenProps): R
     setIsAnswerSubmitted(true)
   }
 
+  useEffect(() => {
+    if (hasCheated) {
+      needsToBeRepeated = false
+      continueExercise().catch(log)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [documentsWithResults])
+
+  const cheatExercise = async (result: SimpleResult): Promise<void> => {
+    setHasCheated(true)
+    setCurrentIndex(documentsWithResults.length - 1)
+
+    const cheatedDocuments = documentsWithResults.map(it => ({ ...it, numberOfTries: 1, result })).reverse()
+    setDocumentsWithResults(cheatedDocuments)
+  }
```

Comment:
I think there is a simpler way to do this :) Hint: Just adapt `finishExercise` to match the `onExerciseFinished` function from the single choice exercises.

Reply:
Good idea, much simpler.

I'll put the time spent on this down as a learning exercise, and from now on remember it's fine to change others' code...

Reply:
Definitely! I mean it worked fine the way you did it as well so it was a perfectly fine solution as well :) And yes, no reason not to change other peoples code if it makes sense :+1: